### PR TITLE
Net 6575- Modify Gateways Cleanup Job to cleanup v2 resources

### PR DESCRIPTION
**Pull request by sarahalsmiller**
Changes proposed in this PR:
- Add v2 resources to gateway cleanup job

How I've tested this PR:
- Local kind helm install/uninstall while running
```kubectl logs job/consul-consul-gateway-cleanup -n consul```
In order to make sure the cleanup job didn't actually error.
- Updated unit test to cover v2 config map method

How I expect reviewers to test this PR:
- Local kind helm install/uninstall from this branch
- CI passes

Checklist:
- [ X ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 1]
```diff
@@ -192,3 +239,89 @@ func exponentialBackoffWithMaxIntervalAndTime() *backoff.ExponentialBackOff {
...
+				return nil
+			}
+			return err
+		}
+
+		// ignore any returned errors
+		_ = c.k8sClient.Delete(context.Background(), gatewayClass)
+
+		// make sure they're gone
+		if err := backoff.Retry(func() error {
```
Comment (jm96441n) [inline control-plane/subcommand/gateway-cleanup/command.go:307]: why do we do this twice? any reason to not just do the entire delete in the retry block to begin with?

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 2]
```diff
@@ -192,3 +239,89 @@ func exponentialBackoffWithMaxIntervalAndTime() *backoff.ExponentialBackOff {
...
+			}
+			return err
+		}
+
+		// ignore any returned errors
+		_ = c.k8sClient.Delete(context.Background(), config)
+
+		// find the gateway class
+		gatewayClass := &v2beta1.GatewayClass{}
+		//TODO is the correct name for the gatewayclass?
```
[the remark this answers is not in the document]
Comment (missylbytes) [reply control-plane/subcommand/gateway-cleanup/command.go:293]: //TODO: NET-6838 To pull GatewayClassName from the Configmap

[the remark this answers is not in the document]
Comment (sarahalsmiller) [reply control-plane/subcommand/gateway-cleanup/command.go:293]: Thank you that makes sense. Will Add.

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 3]
```diff
@@ -8,6 +8,9 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+
```
Comment (missylbytes) [inline control-plane/subcommand/gateway-cleanup/command.go:11]: nit: extra newline

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 4]
```diff
@@ -19,29 +21,44 @@ import (
...
 
+	"github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
 	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
 	"github.com/hashicorp/consul-k8s/control-plane/subcommand"
 	"github.com/hashicorp/consul-k8s/control-plane/subcommand/flags"
 )
 
+const (
+	gatewayConfigFilename  = "/consul/config/config.yaml"
+	resourceConfigFilename = "/consul/config/resources.json"
```
Comment (jm96441n) [inline control-plane/subcommand/gateway-cleanup/command.go:34]: should these live in another package? we use them in the resources job so I'm wondering if they belong in the `gateways` package as constants so we're not defining them twice across this and the resources job

File: control-plane/subcommand/gateway-cleanup/command.go [hunk 5]
```diff
@@ -102,32 +130,59 @@ func (c *Command) Run(args []string) int {
...
+	if err != nil {
+		c.UI.Error(err.Error())
+
+		return 1
+	}
+
+	return 0
+}
+
+func (c *Command) deleteV1GatewayClassAndGatewayClasConfig() error {
```
Comment (jm96441n) [inline control-plane/subcommand/gateway-cleanup/command.go:157]: :raised_hands:

File: control-plane/subcommand/gateway-cleanup/command.go (hunk 4 above)
Comment (sarahalsmiller) [reply control-plane/subcommand/gateway-cleanup/command.go:34]: I'd be in favor of pulling them into a shared package.

**Pull request merged**